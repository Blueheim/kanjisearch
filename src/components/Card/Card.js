import React from 'react';
import classes from './Card.module.scss';
import classNames from 'classnames';

const Card = ({ id, character, meaning, kunyomi, onyomi }) => {
  const cardClasses = classNames('box', 'm-bd-xt', 'm-rd-xt', classes.Card);
  const headerClasses = classNames('m-tx-c');
  const bodyClasses = classNames('side__header', 'm-fx-cl-c-c', classes.Kanji);
  const lectureClasses = classNames('m-fx-cl m-tx-c m-bd-xt-t', classes.Lecture);
  const spellingClasses = classNames('m-wt-900', classes.Spelling);

  const kunLength = kunyomi.length;
  const onLength = onyomi.length;

  const kunyomiDisplay = kunyomi.map((spelling, i) => (
    <div className={spellingClasses} key={i}>
      <span>{spelling}</span>
    </div>
  ));

  const onyomiDisplay = onyomi.map((spelling, i) => (
    <div className={spellingClasses} key={i}>
      <span>{spelling}</span>
    </div>
  ));

  return (
    <div className={cardClasses}>
      <div className="side m-fx-cl">
        <div className={headerClasses}>
          <span>{meaning}</span>
        </div>
        <div className={bodyClasses}>
          <span>{character}</span>
        </div>
        <div className="side__footer m-fx-sh-c">
          <div className={lectureClasses}>
            <div className="m-bd-xt-b">KUN</div>
            <div className="m-fx-sh-c">{kunyomiDisplay}</div>
          </div>
          <div className={lectureClasses}>
            <div className="m-bd-xt-b">ON</div>
            <div className="m-fx-sh-c">{onyomiDisplay}</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
