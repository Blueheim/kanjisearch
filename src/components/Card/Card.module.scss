.Card {
  height: 200px;
}

.Kanji {
  flex-grow: 1;
  font-size: 10rem;
}

.Lecture {
  flex-grow: 1;

  &:not(:last-child) {
    border-right: 1px solid currentColor;
  }
}

.Spelling {
  flex-grow: 1;

  &:not(:last-child) {
    border-right: 1px solid currentColor;
  }
}
