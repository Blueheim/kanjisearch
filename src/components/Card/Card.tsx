import React from 'react';
import classes from './Card.module.scss';
import classNames from 'classnames';
import { IKanji } from '../../types/kanji';

interface ICardStatelessProps {
  id: number;
  character: string;
  meaning: string[];
  kunyomi: string[];
  onyomi: string[];
}

const Card: React.SFC<ICardStatelessProps> = ({ id, character, meaning, kunyomi, onyomi }) => {
  const cardClasses = classNames('box', 'm-bd-xt', 'm-rd-xt', 'm-mg-xt', classes.Card);
  const headerClasses = classNames('m-tx-c', 'm-wt-700', 'm-tertiary');
  const bodyClasses = classNames('side__header', 'm-fx-cl-c-c', 'm-tx-primary', classes.Kanji);
  const lectureClasses = classNames('m-fx-cl m-tx-c m-bd-xt-t', classes.Lecture);
  const spellingClasses = classNames('m-wt-900', classes.Spelling);
  const meaningClasses = classNames(classes.Meaning);

  const meaningDisplay = meaning.map((value: string, i: number) => (
    <span className={meaningClasses} key={i}>
      {value}
    </span>
  ));

  const kunyomiDisplay = kunyomi.map((spelling: string, i: number) => (
    <div className={spellingClasses} key={i}>
      <span>{spelling}</span>
    </div>
  ));

  const onyomiDisplay = onyomi.map((spelling: string, i: number) => (
    <div className={spellingClasses} key={i}>
      <span>{spelling}</span>
    </div>
  ));

  return (
    <div className={cardClasses}>
      <div className="side m-fx-cl">
        <div className={headerClasses}>
          <span>{meaningDisplay}</span>
        </div>
        <div className={bodyClasses}>
          <span>{character}</span>
        </div>
        <div className="side__footer m-fx-sh-c">
          <div className={lectureClasses}>
            <div className="m-bd-xt-b m-primary">KUN</div>
            <div className="m-fx-sh-c m-tx-primary">{kunyomiDisplay}</div>
          </div>
          <div className={lectureClasses}>
            <div className="m-bd-xt-b m-secondary">ON</div>
            <div className="m-fx-sh-c m-tx-secondary">{onyomiDisplay}</div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
