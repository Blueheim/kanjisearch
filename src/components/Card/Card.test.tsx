import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('<Card />', () => {
  const props = {
    id: 1,
    character: '',
    meaning: [],
    kunyomi: [],
    onyomi: [],
  };

  const wrapper = shallow(<Card {...props} />);

  it('expect to render Card component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
