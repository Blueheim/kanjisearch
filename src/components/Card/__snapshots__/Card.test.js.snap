// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> expect to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    character=""
    id={1}
    kunyomi={Array []}
    meaning={Array []}
    onyomi={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="side m-fx-cl"
      >
        <div
          className="m-tx-c m-wt-700 m-tertiary"
        >
          <span />
        </div>
        <div
          className="side__header m-fx-cl-c-c m-tx-primary Kanji"
        >
          <span>
            
          </span>
        </div>
        <div
          className="side__footer m-fx-sh-c"
        >
          <div
            className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
          >
            <div
              className="m-bd-xt-b m-primary"
            >
              KUN
            </div>
            <div
              className="m-fx-sh-c m-tx-primary"
            />
          </div>
          <div
            className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
          >
            <div
              className="m-bd-xt-b m-secondary"
            >
              ON
            </div>
            <div
              className="m-fx-sh-c m-tx-secondary"
            />
          </div>
        </div>
      </div>,
      "className": "box m-bd-xt m-rd-xt m-mg-xt Card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="m-tx-c m-wt-700 m-tertiary"
          >
            <span />
          </div>,
          <div
            className="side__header m-fx-cl-c-c m-tx-primary Kanji"
          >
            <span>
              
            </span>
          </div>,
          <div
            className="side__footer m-fx-sh-c"
          >
            <div
              className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
            >
              <div
                className="m-bd-xt-b m-primary"
              >
                KUN
              </div>
              <div
                className="m-fx-sh-c m-tx-primary"
              />
            </div>
            <div
              className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
            >
              <div
                className="m-bd-xt-b m-secondary"
              >
                ON
              </div>
              <div
                className="m-fx-sh-c m-tx-secondary"
              />
            </div>
          </div>,
        ],
        "className": "side m-fx-cl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span />,
            "className": "m-tx-c m-wt-700 m-tertiary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              
            </span>,
            "className": "side__header m-fx-cl-c-c m-tx-primary Kanji",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
              >
                <div
                  className="m-bd-xt-b m-primary"
                >
                  KUN
                </div>
                <div
                  className="m-fx-sh-c m-tx-primary"
                />
              </div>,
              <div
                className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
              >
                <div
                  className="m-bd-xt-b m-secondary"
                >
                  ON
                </div>
                <div
                  className="m-fx-sh-c m-tx-secondary"
                />
              </div>,
            ],
            "className": "side__footer m-fx-sh-c",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="m-bd-xt-b m-primary"
                  >
                    KUN
                  </div>,
                  <div
                    className="m-fx-sh-c m-tx-primary"
                  />,
                ],
                "className": "m-fx-cl m-tx-c m-bd-xt-t Lecture",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "KUN",
                    "className": "m-bd-xt-b m-primary",
                  },
                  "ref": null,
                  "rendered": "KUN",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "m-fx-sh-c m-tx-primary",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="m-bd-xt-b m-secondary"
                  >
                    ON
                  </div>,
                  <div
                    className="m-fx-sh-c m-tx-secondary"
                  />,
                ],
                "className": "m-fx-cl m-tx-c m-bd-xt-t Lecture",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ON",
                    "className": "m-bd-xt-b m-secondary",
                  },
                  "ref": null,
                  "rendered": "ON",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "m-fx-sh-c m-tx-secondary",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="side m-fx-cl"
        >
          <div
            className="m-tx-c m-wt-700 m-tertiary"
          >
            <span />
          </div>
          <div
            className="side__header m-fx-cl-c-c m-tx-primary Kanji"
          >
            <span>
              
            </span>
          </div>
          <div
            className="side__footer m-fx-sh-c"
          >
            <div
              className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
            >
              <div
                className="m-bd-xt-b m-primary"
              >
                KUN
              </div>
              <div
                className="m-fx-sh-c m-tx-primary"
              />
            </div>
            <div
              className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
            >
              <div
                className="m-bd-xt-b m-secondary"
              >
                ON
              </div>
              <div
                className="m-fx-sh-c m-tx-secondary"
              />
            </div>
          </div>
        </div>,
        "className": "box m-bd-xt m-rd-xt m-mg-xt Card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="m-tx-c m-wt-700 m-tertiary"
            >
              <span />
            </div>,
            <div
              className="side__header m-fx-cl-c-c m-tx-primary Kanji"
            >
              <span>
                
              </span>
            </div>,
            <div
              className="side__footer m-fx-sh-c"
            >
              <div
                className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
              >
                <div
                  className="m-bd-xt-b m-primary"
                >
                  KUN
                </div>
                <div
                  className="m-fx-sh-c m-tx-primary"
                />
              </div>
              <div
                className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
              >
                <div
                  className="m-bd-xt-b m-secondary"
                >
                  ON
                </div>
                <div
                  className="m-fx-sh-c m-tx-secondary"
                />
              </div>
            </div>,
          ],
          "className": "side m-fx-cl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span />,
              "className": "m-tx-c m-wt-700 m-tertiary",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                
              </span>,
              "className": "side__header m-fx-cl-c-c m-tx-primary Kanji",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
                >
                  <div
                    className="m-bd-xt-b m-primary"
                  >
                    KUN
                  </div>
                  <div
                    className="m-fx-sh-c m-tx-primary"
                  />
                </div>,
                <div
                  className="m-fx-cl m-tx-c m-bd-xt-t Lecture"
                >
                  <div
                    className="m-bd-xt-b m-secondary"
                  >
                    ON
                  </div>
                  <div
                    className="m-fx-sh-c m-tx-secondary"
                  />
                </div>,
              ],
              "className": "side__footer m-fx-sh-c",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="m-bd-xt-b m-primary"
                    >
                      KUN
                    </div>,
                    <div
                      className="m-fx-sh-c m-tx-primary"
                    />,
                  ],
                  "className": "m-fx-cl m-tx-c m-bd-xt-t Lecture",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "KUN",
                      "className": "m-bd-xt-b m-primary",
                    },
                    "ref": null,
                    "rendered": "KUN",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                      "className": "m-fx-sh-c m-tx-primary",
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="m-bd-xt-b m-secondary"
                    >
                      ON
                    </div>,
                    <div
                      className="m-fx-sh-c m-tx-secondary"
                    />,
                  ],
                  "className": "m-fx-cl m-tx-c m-bd-xt-t Lecture",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "ON",
                      "className": "m-bd-xt-b m-secondary",
                    },
                    "ref": null,
                    "rendered": "ON",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                      "className": "m-fx-sh-c m-tx-secondary",
                    },
                    "ref": null,
                    "rendered": Array [],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
